combat-trainer.lic

```diff
@@ -55,12 +43,12 @@ class SetupProcess
       game_state.next_clean_up_step
       return true
     end
-    game_state.dancing = should_dance?
-    game_state.now_retreating = should_retreat?
+    game_state.update_dancing(DRRoom.npcs.length)
+    game_state.update_retreating(DRRoom.npcs.length)
```

Comment:
Could we just call game_state.update_npcs(DRRoom.npcs)? and have it deal with these?